### PR TITLE
[IMP] *_queued: Remove `cancel` method + use same user

### PR DESCRIPTION
**Pull request by pedrobaeza**
Thanks to @guewen and @lmignon for their comments.

I'm missing an argument for the method `with_delay` for indicating the user who is going to execute the job (if different from current one).

I can't use `identity_key` in sales order invoicing as you can instruct a sales order to be invoiced in different groupings, having different identity key.

cc @Tecnativa

### CODE REVIEW

**Review by lmignon — APPROVED** [review]
Thank you @pedrobaeza for the changes. This code looks better. (code review only)

**Comment by pedrobaeza** [comment]
Fixed tests + add comments from @guewen from the other place (no need to make all the sudo stuff).

@guewen please continue commenting here any extra thing.

File: account_invoice_validation_queued/models/account_invoice.py [hunk 1]
```diff
@@ -26,6 +26,9 @@ class AccountInvoice(models.Model):
         copy=False,
     )
 
+    def _get_validation_queued_identity_key(self, job):
+        return self.id
```
Comment (guewen) [inline account_invoice_validation_queued/models/account_invoice.py]: The identity key should at least include the model and method names so 2 kind of jobs won't conflict, you can take example on https://github.com/OCA/queue/blob/fdcde048348694df37ce2f63686467cf762aeda5/queue_job/job.py#L102-L144 (or directly use it, I think you want the `identity_exact` here)

Comment (pedrobaeza) [reply account_invoice_validation_queued/models/account_invoice.py]: Thanks for the tip. I have used `identity_exact`.

File: account_invoice_validation_queued/views/queue_job_views.xml [hunk 2]
```diff
@@ -16,7 +16,7 @@
                         class="oe_highlight"
                         attrs="{'invisible': [('state', '!=', 'failed')]}"/>
                  <button type="object"
-                         name="cancel"
+                         name="unlink"
```
Comment (yajo) [inline account_invoice_validation_queued/views/queue_job_views.xml:19]: You can apply this if you want the same functionality as before, with the `cancel()` method.

```suggestion
                         name="unlink"
                         attrs="{'invisible': [('state', 'not in', ('pending', 'enqueued'))]}"
```

File: sale_order_invoicing_queued/views/queue_job_views.xml [hunk 3]
```diff
@@ -17,7 +17,7 @@
                         class="oe_highlight"
                         attrs="{'invisible': [('state', '!=', 'failed')]}"/>
                  <button type="object"
-                         name="cancel"
+                         name="unlink"
```
Comment (yajo) [inline sale_order_invoicing_queued/views/queue_job_views.xml:20]: Here too.

```suggestion
                         name="unlink"
                         attrs="{'invisible': [('state', 'not in', ('pending', 'enqueued'))]}"
```

**Review by yajo — APPROVED** [review]
Indeed it looks much better. :blush: 

A couple of suggestions.

**Review by guewen — APPROVED** [review]
Great, thanks!

File: account_invoice_validation_queued/views/queue_job_views.xml (hunk 2 above)
Comment (pedrobaeza) [reply account_invoice_validation_queued/views/queue_job_views.xml:19]: It's already put 3 lines below

File: sale_order_invoicing_queued/views/queue_job_views.xml (hunk 3 above)
Comment (pedrobaeza) [reply sale_order_invoicing_queued/views/queue_job_views.xml:20]: It's already put 3 lines below

**Comment by pedrobaeza** [comment]
OK, proceeding to merge then. Thanks all for your suggestions.

**Pull request merged**